Data/Functor/Foldable/TH.hs

```diff
@@ -0,0 +1,141 @@
+{-# LANGUAGE TemplateHaskell #-}
```

Comment:
As a general rule, @ekmett likes to keep his libraries free of the `TemplateHaskell` extension so that they can build on stage-1 compilers and the like. In this case, it should be possible to avoid the use of `TemplateHaskell` relatively easily. Just replace any quoted `Name` with something like [this](https://github.com/ekmett/lens/blob/7dadc58cdc55bd00e17a14bc97c60e354a5cbd11/src/Control/Lens/Internal/TH.hs#L95-L116) (example from `lens`).